networkx/algorithms/flow/networksimplex.py

```diff
@@ -540,7 +546,7 @@ def update_potentials(i, p, q):
     # Infeasibility and unboundedness detection
     ###########################################################################
 
-    if any(x[i] != 0 for i in range(-n, 0)):
+    if any(abs(x[i]) > tolerance for i in range(-n, 0)):
```

Comment:
Why not just `if any(abs(d) > tolerance for d in x):`. (Or change `d` to whatever variable name makes sense for the elements of `x`.)